src/app/core/store/documents/documents.effects.ts

```diff
@@ -287,31 +286,32 @@ export class DocumentsEffects {
 function createSyncCollectionAction(collection: CollectionModel,
                                     newDocument: DocumentModel,
                                     oldDocument: DocumentModel): CollectionsAction.UpdateSuccess {
-  const newAttributeIds: string[] = newDocument && newDocument.data ? Object.keys(newDocument.data) : [];
-  const oldAttributeIds: string[] = oldDocument && oldDocument.data ? Object.keys(oldDocument.data) : [];
+  const newAttributeNames: string[] = newDocument && newDocument.data ? Object.keys(newDocument.data) : [];
+  const oldAttributeNames: string[] = oldDocument && oldDocument.data ? Object.keys(oldDocument.data) : [];
 
-  const attributes = updateAttributes(collection.attributes, newAttributeIds, oldAttributeIds);
+  const attributes = updateAttributes(collection.attributes, newAttributeNames, oldAttributeNames);
   const documentsCount = collection.documentsCount + (!oldDocument ? 1 : 0) - (!newDocument ? 1 : 0);
   const updatedCollection: CollectionModel = {...collection, attributes, documentsCount};
 
   return new CollectionsAction.UpdateSuccess({collection: updatedCollection});
 }
 
 function updateAttributes(attributes: AttributeModel[],
-                          newDocumentAttributeIds: string[],
-                          oldDocumentAttributeIds: string[]): AttributeModel[] {
-  const addedAttributeIds = newDocumentAttributeIds.filter(id => !oldDocumentAttributeIds.includes(id));
-  const removedAttributeIds = oldDocumentAttributeIds.filter(id => !newDocumentAttributeIds.includes(id));
+                          newDocumentAttributeNames: string[],
+                          oldDocumentAttributeNames: string[]): AttributeModel[] {
+  const addedAttributeNames = newDocumentAttributeNames.filter(name => !oldDocumentAttributeNames.includes(name));
+  const removedAttributeNames = oldDocumentAttributeNames.filter(name => !newDocumentAttributeNames.includes(name));
 
-  const attributeIds = attributes.map(attribute => attribute.id);
-  const createdAttributes = addedAttributeIds.filter(id => !attributeIds.includes(id))
-    .map(id => ({id, name: extractAttributeName(id), constraints: [], usageCount: 1}));
+  const attributeNames = attributes.map(attribute => attribute.name);
+  const createdAttributes = addedAttributeNames.filter(name => !attributeNames.includes(name))
+    .map(name => ({id: "?????", name, constraints: [], usageCount: 1}));
+  // TODO I don't have attribute id !!!!!!!!!!!!
```

Comment:
As we discussed, we should always create attributes in a collection first and the add some data to these attributes in a document. This can be done as a series of actions (in case of post-it perspective) so the behavior stays the same for the user and it seems like it is created at the same time.